tests/python/relay/test_any.py

```diff
@@ -1418,10 +1418,14 @@ def _body(i, st):
     start = relay.var("start", shape=(), dtype="int32")
     body = loop(start, relay.op.reshape(relay.const(0), newshape=(1, 1)))
     func = relay.Function([start], relay.TupleGetItem(body, 1))
+
     with DiagnosticTesting() as diagnostics:
         diagnostics.assert_message(
-            "The Relay type checker is unable to show the following types "
-            "match.\nIn particular dimension 0 conflicts: 2 does not match 1."
+            "The Relay type checker is unable to show the following types match:\n"
+            "  Tensor[(2, 1), int32]\n"
+            "  Tensor[(1, 1), int32]\n"
```

Comment:
What is this change?

Reply:
I ran into this error message at some point and found it pretty inscrutable without formatting + some extra info (the changes to `type_solver.cc`). This test verified the error message so it had to be updated as well. I lumped it in with `mv` since that's where I ran into it but I can break it out into its own PR if necessary

Reply:
oh I missed that `diagnostics.assert_message`. No need for a separate PR